lib/src/util.dart

```diff
@@ -1,4 +1,72 @@
 import 'dart:convert';
 
+import 'package:charcode/charcode.dart';
+
 String escapeHtml(String html) =>
     const HtmlEscape(HtmlEscapeMode.ELEMENT).convert(html);
+
+// Escape the contents of [value], so that it may be used as an HTML attribute.
+
+// Based on http://spec.commonmark.org/0.28/#backslash-escapes.
+String escapeAttribute(String value) {
```

Comment:
This seems like a common algorithm... is it really not in a shared package somewhere? If not, it may be worth adding it to `http_parser` or something.

Reply:
Not with the weird backslash escaping as well. I think `dart:convert` has the method we want, after I move the backslash escaping into the parsing/string-buffer-building code.